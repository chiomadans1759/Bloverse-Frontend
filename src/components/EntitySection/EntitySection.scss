#entity-section {
  overflow-x: hidden;
  overflow-y: auto;
  position: relative;
  bottom: 0;
  height: 65%;
  padding-bottom: 1rem;
  .item {
    img {
      -o-object-fit: cover;
      object-fit: cover;
      transition: all cubic-bezier(0.075, 0.82, 0.165, 1);
    }
    .count {
      display: none;
    }
    &.active img {
      width: 5rem;
      height: 5rem;
    }
    .count {
      svg {
        color: #b7b7b7 !important;
        margin-right: 0.25rem;
      }
      .divider-circle {
        margin: 0 0.5rem;
        width: 0.25rem;
        height: 0.25rem;
        background: #096dd9;
        border-radius: 100%;
        display: inline-block;
      }
    }
    &.active .count {
      display: block;
      margin-top: 0.5rem;
    }
    img {
      width: 2.5rem;
      height: 2.5rem;
    }
  }
  &::-webkit-scrollbar-track {
    background-color: transparent;
    border-radius: 0.5rem;
  }
  &::-webkit-scrollbar {
    width: 0.375rem;
    background-color: transparent;
  }
  &::-webkit-scrollbar-thumb {
    background-color: #000000;
  }
}
