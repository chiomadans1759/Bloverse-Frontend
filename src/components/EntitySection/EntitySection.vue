<template>
  <main id="entity-section" class="scrollbar">
    <div class="d-flex align-items-center mb-4 item active">
      <img src="./../../assets/category/category3.jpg" class="rounded-circle mr-3">
      <div>
          <p class="m-0">Entity Name</p>
          <div class="count d-flex align-items-center">
            <span><i class="fal fa-id-card"></i>758</span>
            <span class="divider-circle"></span>
            <span><i class="fal fa-comment-dots"></i>234</span>
          </div>
      </div>      
    </div>
    <div class="mb-3" v-for="i in 7" :key="i">
      <div class="item d-flex align-items-center ">
        <img src="./../../assets/category/category7.jpg" class="rounded-circle mr-3">
        <div class="scroll">
            <p class="m-0">Entity Name</p>
            <div class="count">
              <span><i class="fal fa-id-card"></i>758</span>
              <span class="circle mx-2"></span>
              <span><i class="fal fa-comment-dots"></i>234</span>
            </div>
        </div> 
      </div>
    </div>
  </main>
</template>

<script>
export default {
  
}
</script>

<style lang="scss" scoped>
  @import './EntitySection'
</style>
