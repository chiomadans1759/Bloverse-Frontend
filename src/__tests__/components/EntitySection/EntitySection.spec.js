import { mount } from '@vue/test-utils';
import EntitySection from '@/components/EntitySection/EntitySection.vue';

describe('EntitySection Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(EntitySection);

    expect(wrapper.element).toMatchSnapshot();
  });
});
