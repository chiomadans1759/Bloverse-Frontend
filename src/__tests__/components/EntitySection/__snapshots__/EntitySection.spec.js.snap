// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntitySection Component should render correctly 1`] = `
<main
  class="scrollbar"
  id="entity-section"
>
  <div
    class="d-flex align-items-center mb-4 item active"
  >
    <img
      class="rounded-circle mr-3"
      src="./../../assets/category/category3.jpg"
    />
     
    <div>
      <p
        class="m-0"
      >
        Entity Name
      </p>
       
      <div
        class="count d-flex align-items-center"
      >
        <span>
          <i
            class="fal fa-id-card"
          />
          758
        </span>
         
        <span
          class="divider-circle"
        />
         
        <span>
          <i
            class="fal fa-comment-dots"
          />
          234
        </span>
      </div>
    </div>
  </div>
   
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-3"
  >
    <div
      class="item d-flex align-items-center "
    >
      <img
        class="rounded-circle mr-3"
        src="./../../assets/category/category7.jpg"
      />
       
      <div
        class="scroll"
      >
        <p
          class="m-0"
        >
          Entity Name
        </p>
         
        <div
          class="count"
        >
          <span>
            <i
              class="fal fa-id-card"
            />
            758
          </span>
           
          <span
            class="circle mx-2"
          />
           
          <span>
            <i
              class="fal fa-comment-dots"
            />
            234
          </span>
        </div>
      </div>
    </div>
  </div>
</main>
`;
